delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/Storages.scala

```diff
@@ -111,10 +110,10 @@ final class Storages private (
       id: IdSegment,
       projectRef: ProjectRef,
       storageFields: StorageFields
-  )(implicit caller: Caller): IO[StorageRejection, StorageResource] = {
+  )(implicit caller: Caller): IO[StorageResource] = {
     for {
-      pc    <- fetchContext.onCreate(projectRef)
-      iri   <- expandIri(id, pc)
+      pc    <- fetchContext.onCreate(projectRef).toCatsIO
+      iri   <- expandIri(id, pc).toCatsIO
```

Comment:
As an aside this `ExpandIri` could definitely be refactored, probably once the migration is done.  It doesn't do any actual effects - I prefer using less powerful types like `Either`/`Option` with the translation to the effect system delayed until the call sites. It makes things easier to understand / refactor / test IMO! E.g. in this case all it's doing is moving between data types.

This might be a larger larger discussion though? Sort of a stylistic preference

Reply:
Yeah I suppose I'm not clear why it's an IO

Reply:
I agree the change you made was nice but I'm going to be lazy and not fix this